                Вложенный оператор try.
    Операторы try могут быть вложенными. Это означает, что один оператор try
может находиться в блоке друтого оператора try. Всякий раз, когда управление
передается блоку оператора try, контекст соответствующего исключения раз­ме-
щается в стеке. Если во вложенном операторе try отсутствует оператор catch
для перехвата и обработки конкретного исключения, стек развертывается и прове-
ряется на соответствие оператор catch из внешнего блока оператора try, и так
до тех пор, пока не будет найден подходящий оператор catch или не будут исчер-
паны все уровни вложенности операторов try. Если подходящий оператор catch не
будет найден, то возникшее исключение обработает исполняющая система Java.

Пример 1.
    В этой программе один блок try вложен в другой. Программа ра­ботает следующим
образом. Когда она запускается на выполнение без аргументов командной строки, во
внешнем блоке оператора try генерируется исключение в связи с делением на нуль. А
если программа запускается на выполнение с одним аргументом, то исключение в связи
с делением на нуль генерируется во вложен­ном блоке оператора try. Но поскольку это
исключение не обрабатывается во вложенном блоке, то оно передается внешнему блоку
оператора try, где и обрабатывается. Если же программе передаются два аргумента
командной строки, то во внутреннем блоке оператора try генерируется исключение в
связи с выходом индекса за пределы массива.

C:\>java NestTry
Деление на нуль: java.lang.ArithmeticException: / bу zero

C:\>java NestTry One
а=1
Деление на нуль: java.lang.ArithmeticException: / bу zero

C : \>java NestTry One Two
а=2
Индекс за пределами массива: java.lang.ArrayIndexOutOfBoundsException:42